File: 3334-apple-redistribution-into-boxes/solution.cpp
#include <vector>
#include <numeric>
#include <algorithm>

class Solution {
public:
    int minimumBoxes(std::vector<int>& apple, std::vector<int>& capacity) {
        int totalApples = std::accumulate(apple.begin(), apple.end(), 0);

        std::sort(capacity.begin(), capacity.end(), std::greater<int>());
        
        int boxes = 0;
        for (int cap : capacity) {
            totalApples -= cap;
            boxes++;
            if (totalApples <= 0) break;
        }
        return boxes;
    }
};

File: 3334-apple-redistribution-into-boxes/solution.java
class Solution 
{
    public int minimumBoxes(int[] apple, int[] capacity) 
    {
        int total = 0 ;
        int cnt = 0 ;

        for(int i=0 ; i<apple.length ; i++)
        {
            total =total + apple[i] ;
        }

        Arrays.sort(capacity) ;

        for(int i=capacity.length-1 ; i>=0  ; i--)
        {
            if(total == 0) break ; 
            if(capacity[i] <= total && total >0)
            {
                cnt++;
                total= total - capacity[i] ;
            }
            else
            {
                cnt++ ;
                total = 0 ;
            }
        }   
        return cnt ;
    }
}
